src/snapred/backend/service/CalibrationService.py

```diff
@@ -216,39 +218,41 @@ def loadQualityAssessment(self, request: CalibrationLoadAssessmentRequest):
         if request.checkExistent:
             wkspaceExists = False
             for metricName in ["sigma", "strain"]:
-                ws_name = (
-                    wng.diffCalMetrics()
-                    .runNumber(request.runId)
-                    .version(request.version)
-                    .metricName(metricName)
-                    .build()
-                )
-                if self.dataFactoryService.workspaceDoesExist(ws_name):
+                wsName = wng.diffCalMetric().metricName(metricName).runNumber(runId).version(version).build()
+                if self.dataFactoryService.workspaceDoesExist(wsName):
                     wkspaceExists = True
                     break
             if not wkspaceExists:
-                for ws_name in calibrationRecord.workspaceNames:
-                    if self.dataFactoryService.workspaceDoesExist(ws_name):
+                for wsName in calibrationRecord.workspaces.values:
+                    wsName += "_" + wnvf.formatVersion(version)
+                    if self.dataFactoryService.workspaceDoesExist(wsName):
                         wkspaceExists = True
                         break
             if wkspaceExists:
                 errorTxt = (
                     f"Calibration assessment for Run {runId} Version {version} "
-                    f"is already loaded: see workspace {ws_name}."
+                    f"is already loaded: see workspace {wsName}."
                 )
                 logger.error(errorTxt)
-                raise ValueError(errorTxt)
+                raise RuntimeError(errorTxt)
 
         # generate metrics workspaces
         GenerateCalibrationMetricsWorkspaceRecipe().executeRecipe(
             CalibrationMetricsWorkspaceIngredients(calibrationRecord=calibrationRecord)
         )
 
-        # load persistent workspaces
-        for ws_name in calibrationRecord.workspaceNames:
-            self.dataFactoryService.getCalibrationDataWorkspace(
-                calibrationRecord.runNumber, calibrationRecord.version, ws_name
-            )
+        # load persistent data workspaces, assuming all workspaces are of WNG-type
+        for wsKey, wsName in calibrationRecord.workspaces.items():
```

Comment:
I understand why is has to be here but would like to look at refactoring it once we have a better way.  Better integration with `WorkspaceName` may help groceryClerk infer things.

(I think technically the `self.groceryClerk.name` can be dragged to the top of the loop as  `self.groceryClerk.name(wsKey)`)

Reply:
I moved the key to the top.  Because of Malcolm's recent request, I also had to break out a temporary fixup for the diffraction-calibration output-workspace "unit" treatment.